### PR TITLE
feat: refactor router

### LINKED ISSUE #151: Items are displayed twice sometime

### PR DESCRIPTION
**Pull request by lonk**
Fixes #151 and #167

### CODE REVIEW

File: src/lib/randString.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export function randString(x) {
```
Comment (gjuchault) [inline src/lib/randString.js]: Can you use lodash here ?
https://runkit.com/585565223d51f5001398c039/5901e9ef1ce99600127a7c0f

File: src/main.js [hunk 2]
```diff
@@ -47,89 +58,59 @@ Vue.component('b-table', PaginatedTable);
...
+                    value: { id: route.params.id }
+                })
+                .then(() => next())
+                .catch(() => next(from.path));
+            } else {
+                store.dispatch('clearModObject');
+                next();
+            }
+        })
+        .catch(() => store.dipatch('showClientError', 'Impossible de charger l\'interface'));
```
Comment (gjuchault) [inline src/main.js]: More explicit error message ? Missing rights ?

File: src/store/modules/objects.js [hunk 3]
```diff
@@ -23,172 +23,21 @@ const mutations = {
...
-
-        state_.events.splice(i, 1);
-    },
-    DELETESET(state_, set) {
-        const i = state_.sets.findIndex(s => (s.id === set.id));
+    UPDATEOBJECT(state_, data) {
+        const i = state_[data.route].findIndex(o => (o.id === data.object.id));
 
-        state_.sets.splice(i, 1);
+        state_[data.route][i] = Object.assign(state_[data.route][i], data.object);
```
Comment (gjuchault) [inline src/store/modules/objects.js]: `Object.assign` actually change the object (pointers).
So either:
`Object.assign(state_[data.route][i], data.object);`
Or the safest way (choose this one 👍):
`state_[data.route][i] = Object.assign({}, state_[data.route][i], data.object);`

File: src/main.js (hunk 2 above)
Comment (lonk) [inline src/main.js]: The only way this error can occur is if the server can't be reached

**Pull request merged**